MinVer/MinVer.csproj

```diff
@@ -40,7 +42,12 @@
     <None Remove="buildMultiTargeting\**\*" />
     <Content Include="build\**\*" PackagePath="build" />
     <Content Include="buildMultiTargeting\**\*" PackagePath="buildMultiTargeting" />
-    <Content Include="$(OutputPath)\publish\**\*" PackagePath="minver" Visible="false" Link="hidden" />
   </ItemGroup>
 
+  <Target Name="AddMinVerOutput" BeforeTargets="_GetPackageFiles">

```

Comment:
```text
BeforeTargets="_GetPackageFiles"
```

This is the key thing. I think I'll never understand how you find these things! 😆

Reply:
Strictly speaking that target just needs to run after all the artifacts are compiled and written to disk, so I could have used something like `AfterTargets="CoreCompile"` instead. However, since `_GetPackageFiles` is the actual target that does the work of turning the various build action item groups into the list of files that need to be packaged, I decided to use that one instead.

Reply:
> since `_GetPackageFiles` is the actual target that does the work of turning the various build action item groups into the list of files that need to be packaged

How did you find that out?

Reply:
> How did you find that out?

The key tools for any sort of MSBuild investigation is to get a binary log, which is easy to do in VS with the [Project System Tools](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools) extension), and then analyze that log file with the [MSBuild Structured Log Viewer](https://msbuildlog.com/).

From there, it's possible to see the actual targets being run and actually open up the targets files and look at what they are doing.

I've spent a lot of time looking at the actual NuGet packaging targets files in the past, so I was already aware that there was a target that did the conversion, but I did have to look at the log file to rediscover the name of it.

I'd be happy to walk you through my process for this sort of thing if you're interested.

Reply:
Thanks, I grabbed Project System Tools (and updated my installation of MSBuild Structured Log Viewer) and I managed to examine the build log for the MinVer project.

Still, there is a _whole_ lot going on between `CoreCompile` and `_GetPackageFiles`, and I find the `targets` source files very hard to read. I'm not sure that my workflow for identifying where to inject my target would ever be much more than trial and error. 🤷

![image](https://user-images.githubusercontent.com/677704/103996305-ed6b2400-5199-11eb-9516-9e25319efe87.png)